adaptors_task/student/src/impl.cpp

```diff
@@ -1,8 +1,67 @@
 #include "impl.h"
 #include <vector>
+#include <stack>
+#include <tuple> 
 
 const std::vector<std::pair<char, char>> brackets {
     {'(', ')'},
     {'[', ']'},
     {'{', '}'}
 };
+
+enum class KindOfChar {OpenBracket, ClosedBracket, NoBracket};
+
+std::pair<KindOfChar,  char> Find_KindOfChar_CorrespondingBracket(const char & c){
+    for(const auto & b : brackets ){
+         if(c==b.first){
```

Comment:
Instead of writing comments, I'd also suggest to add some hints for the reader with help of const bools
if (const isOpenBracket = (sourceChar == bracket_pair.first))

nothing is changed, but condition now have a name